### PR TITLE
Bugfix/OP-588 - fix to mimetype checker

### PR DESCRIPTION
**Pull request by aot221**
[OP-588](https://nycrecords.atlassian.net/browse/OP-588)

### CODE REVIEW

File: public_records_portal/upload_helpers.py [hunk 1]
```diff
@@ -23,17 +23,40 @@
                       'ppsx', 'xlsx', 'jpg', 'jpeg', 'png', 'gif', 'tif',
```
Comment (joelbcastillo) [inline public_records_portal/upload_helpers.py]: If this isn't being used, can we remove it from the code?

File: public_records_portal/upload_helpers.py [hunk 2]
```diff
@@ -234,7 +257,14 @@ def upload_file_locally(document, filename, privacy, request_id = None):
 
 ### @export "allowed_file"
 def allowed_file(file, request_id, privacy=1):
-    mimetypeMagic = magic.Magic(mime=True,uncompress=True)
-    mimetype = mimetypeMagic.from_buffer(file.read())
+    '''Checks the mimetype of the file to see if its allowed'''
```
Comment (joelbcastillo) [inline public_records_portal/upload_helpers.py]: Please update the docstring to include a description of the parameters and return type.

File: public_records_portal/views.py [hunk 3]
```diff
@@ -666,11 +666,15 @@ def upload_document():
     for file in files:
         document_upload_errors = {}
         secure_fname = secure_filename(file.filename)
-        document_filepath = app.config["UPLOAD_PUBLIC_LOCAL_FOLDER"] + '/' + form['request_id'] + '/' + secure_fname
-        doc_id, filename, errors = upload_file(file, form['request_id'], 0x1)
-        if (os.path.exists(document_filepath)):
+        # Searches through public and private folders for any pre-existing documents
```
Comment (joelbcastillo) [inline public_records_portal/views.py]: Why don't we just search in the database to see if a document with the same name exists for the current request?

File: public_records_portal/views.py [hunk 4]
```diff
@@ -666,11 +666,15 @@ def upload_document():
...
+        document_filepath_public = app.config["UPLOAD_PUBLIC_LOCAL_FOLDER"] + '/' + form['request_id'] + '/' + secure_fname
+        document_filepath_private = app.config["UPLOAD_PRIVATE_LOCAL_FOLDER"] + '/' + form['request_id'] + '/' + secure_fname
+        if (os.path.exists(document_filepath_public)):
+            document_upload_errors['duplicate'] = 'There already eixsts a document of the same name.'
+        elif (os.path.exists(document_filepath_private)):
             document_upload_errors['duplicate'] = 'There already eixsts a document of the same name.'
         else:
+            doc_id, filename, errors = upload_file(file, form['request_id'], 0x1)
             if doc_id == 'VIRUS_FOUND':
```
Comment (joelbcastillo) [inline public_records_portal/views.py]: This function returns None if a virus is found, not the string "VIRUS_FOUND"

File: public_records_portal/upload_helpers.py [hunk 5]
```diff
@@ -232,9 +255,20 @@ def upload_file_locally(document, filename, privacy, request_id = None):
...
+    ms = magic.open(magic.NONE)
+    ms.load()
+    mimetype = ms.buffer(file.read())
+    app.logger.info("\n\nMimetype: " + mimetype)
     file.seek(0)
-    return mimetype in ALLOWED_MIMETYPES
\ No newline at end of file
+    # Loops through the ALLOWED_MIMETYPES list and checks if the file's mimetype is inside
+    for m in ALLOWED_MIMETYPES:
+        if m in mimetype:
```
Comment (ezliang) [inline public_records_portal/upload_helpers.py]: should be if m == mimetype

**Pull request merged**